ClickhouseFile sink时报错

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

ClickhouseFile sink时报错

### SeaTunnel Version

2.3.11

### SeaTunnel Config

```conf
env {
	job.name = "mysql-to-clickhousefile"
	format = "hocon"
	job.mode = "BATCH"
}

source {
	Jdbc {
		url = "jdbc:mysql://xxxxx:3306/db_name?useUnicode=true&characterEncoding=utf8&zeroDateTimeBehavior=convertToNull&useSSL=true&serverTimezone=GMT%2B8"
		driver = "com.mysql.cj.jdbc.Driver"
		connection_check_timeout_sec = 100
		user = "xxx"
		password = "xxx"
		query = "select * from xxx.xxx"
	}
}

transform {
}

sink {
	ClickhouseFile {
		host = "xxx:xxx"
		database = "test"
		table = "seatunnel_write_file"
		username = "default"
		password = "xxx"
		clickhouse_local_path = "/home/seatunnel/clickhouse/data"
		node_free_password = true
		node_pass = [{
		    node_address = "xxx"
		    password = "xxx"
	     }]
	}
}
```

### Running Command

```shell
rest api
```

### Error Exception

```log
Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:228)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:40)
Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize failed] - Unable to create a sink for identifier 'ClickhouseFile'.
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSink(FactoryUtil.java:250)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.createSinkAction(MultipleTableJobConfigParser.java:669)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSink(MultipleTableJobConfigParser.java:592)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:240)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:123)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:191)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:165)
        ... 2 more
Caused by: java.lang.IllegalArgumentException: Json parsing exception, value '[{node_address=xxxxx, password=xxxxx}]', and expected type 'org.apache.seatunnel.api.configuration.Options$OptionBuilder$13$1@7cca01a8'
        at org.apache.seatunnel.api.configuration.util.ConfigUtil.convertValue(ConfigUtil.java:78)
        at org.apache.seatunnel.api.configuration.ReadonlyConfig.getOptional(ReadonlyConfig.java:123)
        at org.apache.seatunnel.api.configuration.ReadonlyConfig.get(ReadonlyConfig.java:66)
        at org.apache.seatunnel.connectors.seatunnel.clickhouse.sink.file.ClickhouseFileSinkFactory.createSink(ClickhouseFileSinkFactory.java:142)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSink(FactoryUtil.java:247)
        ... 8 more
Caused by: org.apache.seatunnel.shade.com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "node_address" (class org.apache.seatunnel.connectors.seatunnel.clickhouse.config.NodePassConfig), not marked as ignorable (2 known properties: "nodeAddress", "password"])
 at [Source: (String)"[{"node_address":"192.168.3.203","password":"behome"}]"; line: 1, column: 19] (through reference chain: java.util.ArrayList[0]->org.apache.seatunnel.connectors.seatunnel.clickhouse.config.NodePassConfig["node_address"])
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1127)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:2023)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1700)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1678)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:319)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:176)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:355)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:244)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:28)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4674)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3629)
        at org.apache.seatunnel.shade.com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3612)
        at org.apache.seatunnel.api.configuration.util.ConfigUtil.convertValue(ConfigUtil.java:58)
        ... 12 more
```

### Zeta or Flink or Spark Version

zeta

### Java or Scala Version

java8

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/config/NodePassConfig.java
+++ b/seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/config/NodePassConfig.java
@@ -17,6 +17,9 @@
 
 package org.apache.seatunnel.connectors.seatunnel.clickhouse.config;
 
+import org.apache.seatunnel.shade.com.fasterxml.jackson.annotation.JsonAlias;
+import org.apache.seatunnel.shade.com.fasterxml.jackson.annotation.JsonProperty;
+
 import org.apache.seatunnel.api.configuration.util.OptionMark;
 
 import lombok.Data;
@@ -24,8 +27,18 @@ import lombok.Data;
 @Data
 public class NodePassConfig {
 
+    // The documented config key is snake_case ("node_pass.node_address"), while
+    // ReadonlyConfig converts the value with a plain ObjectMapper that has no
+    // snake_case naming strategy. Without the explicit mapping the conversion
+    // fails with an UnrecognizedPropertyException and the sink cannot be
+    // created (issue #9889).
     @OptionMark(description = "The address of Clickhouse server node")
+    @JsonProperty("node_address")
+    @JsonAlias("nodeAddress")
     private String nodeAddress;
+
+    @OptionMark(description = "Clickhouse server linux username")
+    private String username;
 
     @OptionMark(description = "Clickhouse server linux password")
     private String password;

--- a/seatunnel-connectors-v2/connector-clickhouse/src/test/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/config/NodePassConfigTest.java
+++ b/seatunnel-connectors-v2/connector-clickhouse/src/test/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/config/NodePassConfigTest.java
@@ -1,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.clickhouse.config;
+
+import org.apache.seatunnel.api.configuration.ReadonlyConfig;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class NodePassConfigTest {
+
+    @Test
+    public void testNodePassConfigParsesDocumentedSnakeCaseKeys() {
+        // The documented node_pass entries use snake_case keys
+        // (node_pass.node_address / node_pass.username / node_pass.password).
+        // ReadonlyConfig.get(NODE_PASS) must convert them into NodePassConfig;
+        // before the fix the plain ObjectMapper conversion failed with
+        // UnrecognizedPropertyException and the sink could not be created
+        // (issue #9889).
+        Map<String, Object> nodePass = new HashMap<>();
+        nodePass.put("node_address", "10.0.0.1:8123");
+        nodePass.put("username", "default");
+        nodePass.put("password", "secret");
+
+        ReadonlyConfig config =
+                ReadonlyConfig.fromMap(
+                        Collections.singletonMap(
+                                ClickhouseFileSinkOptions.NODE_PASS.key(),
+                                Collections.singletonList(nodePass)));
+
+        List<NodePassConfig> parsed = config.get(ClickhouseFileSinkOptions.NODE_PASS);
+        Assertions.assertEquals(1, parsed.size());
+        Assertions.assertEquals("10.0.0.1:8123", parsed.get(0).getNodeAddress());
+        Assertions.assertEquals("default", parsed.get(0).getUsername());
+        Assertions.assertEquals("secret", parsed.get(0).getPassword());
+    }
+
+    @Test
+    public void testNodePassConfigAcceptsCamelCaseAlias() {
+        Map<String, Object> nodePass = new HashMap<>();
+        nodePass.put("nodeAddress", "10.0.0.2:8123");
+        nodePass.put("password", "secret2");
+
+        ReadonlyConfig config =
+                ReadonlyConfig.fromMap(
+                        Collections.singletonMap(
+                                ClickhouseFileSinkOptions.NODE_PASS.key(),
+                                Collections.singletonList(nodePass)));
+
+        List<NodePassConfig> parsed = config.get(ClickhouseFileSinkOptions.NODE_PASS);
+        Assertions.assertEquals(1, parsed.size());
+        Assertions.assertEquals("10.0.0.2:8123", parsed.get(0).getNodeAddress());
+        Assertions.assertEquals("secret2", parsed.get(0).getPassword());
+    }
+}

